### DISCUSSION TITLE
Google Analytics

### DISCUSSION THREAD

**Discussion by TiCampbell** [Get-Your/Get-Your#110] [General]
@penoptech Are you very familiar with Google Analytics? I'm just getting into its advanced features since I got an email notifying that Universal Analytics is being replaced with Google Analytics 4.

Based on my cursory exploration, I'd like to separate DEV, STAGE, and PROD apps into separate Data Streams (so we could look at user analytics just for actual users and ignore test users or developers). What do you think?

It seems like it would be straightforward to do this with application settings within each Azure Web App deployment slot - I could create a variable for each environment and set it to the Measurement ID of the proper Stream, which would then be injected into the container as an environment variable. What would the difficulty be of changing the `gtag` values to use environment variables and (probably more importantly) would the app have issues with a blank or nonexistent `gtag` value when run locally?

**Comment by penoptech** [comment]
@TiCampbell - What I've done before and something you've already set us up to do, is conditionally render the script tags necessary for Google Analytics. Something like this:

```
    {% if HOST_ENVIRONMENT == 'production'%}
    <!-- Global site tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=<<your_ua_id_here>>"></script>
    <script>
        window.dataLayer = window.dataLayer || [];

        function gtag() {
            dataLayer.push(arguments);
        }
        gtag("js", new Date());

        gtag("config", "UA-167996300-1");
    </script>
    {% endif %}
``` 

So you could have just have the analytics for the production environment and not worry about the having it for other environments, like DEV or STAGE.
